### PR TITLE
Kieker 1959 add oceandsl features

### PR DESCRIPTION
**Pull request by rju**
# Pull Request

## Contribution
This pull request provides the following contribution to Kieker
- Integrated all oceandsl tools
- Removed the generation of ZIP-Archive of all, as this is too big now for everything. Better packaging is the goal of another ticket.

## Code Quality Thresholds
- Increased do to huge code influx and a lot of additional tools.

### CODE REVIEW

**Comment by rju** [comment]
Die Builds auf Jenikns sind erfolgreich durchgelaufen. Es meckert natürlich über einige fehlgeschlagene Tests. Das liegt daran, dass einige Stages von OceanDSL zwar Tests haben, diese aber noch nicht implementiert sind. Das würde ich aber ungern in diesem Ticket erledigen.

File: gradle.properties [hunk 1]
```diff
@@ -7,12 +7,15 @@ dirConfig = config
 dirLib = lib
 dirQuality = staticAnalysis
 
-org.gradle.jvmargs=-Xms256m -Xmx2048m
```
Comment (DaGeRe) [inline gradle.properties:10]: Why was it necessary to increase the RAM? If its necessary, I'd recommend to just increase it and not leave the old one commented

File: gradle.properties [hunk 2]
```diff
@@ -7,12 +7,15 @@ dirConfig = config
...
+# org.gradle.jvmargs=-Xms256m -Xmx2048m
+org.gradle.jvmargs=-Xms2560m -Xmx8096m
 
 # analysis properties
 checkstyleVersion = 8.35
 ## Updating to 6.37.0 lead to a timeout when publishing -- update when the bug in pmd is fixed
 pmdVersion = 6.25.0
+#pmdVersion = 7.0.0-rc4
```
Comment (DaGeRe) [inline gradle.properties:17]: The commented pmdVersion doesn't make sense

File: Jenkinsfile [hunk 3]
```diff
@@ -31,7 +31,7 @@ pipeline {
     stage('Default Docker Stages') {
       agent {
         docker {
-          image 'kieker/kieker-build:openjdk8'
+          image 'kieker/kieker-build:openjdk11'
```
Comment (DaGeRe) [inline Jenkinsfile:34]: Building everything with JDK 11 will lead to Bytebuffer problems, making also kieker-monitoring effectively unusable with JDK 8 (https://stackoverflow.com/a/61267496/2096209). Therefore, for keeping compatibility of monitoring with JDK, and at the same time compile analysis with JDK 11, we need to separate JDKs.

File: Jenkinsfile [hunk 4]
```diff
@@ -98,7 +98,7 @@ pipeline {
 
         stage('Distribution Build') {
           steps {
-            sh './gradlew -x test -x sign build publishToMavenLocal distribute'
+            sh './gradlew -x test -x signMavenJavaPublication build publishToMavenLocal distribute'
```
Comment (DaGeRe) [inline Jenkinsfile:101]: Why is this renamed?

File: build.gradle [hunk 5]
```diff
@@ -810,29 +840,59 @@ def exampleJavaEEContent = project.copySpec {
...
-  ':kieker-tools:log-replayer:distZip',
-  ':kieker-tools:resource-monitor:distZip',
-  ':kieker-tools:trace-analysis:distZip',
-  ':kieker-tools:trace-analysis-gui:distZip',
-  ':kieker-examples:jar',
-  ':kieker-examples:spotlessMisc' ]) {
-  archiveClassifier = 'binaries'
-  with binaryContent, toolsContent, libraryContent, exampleContent, exampleJavaEEContent
-}
+//task distributeBinaryZip(type: Zip, group: 'distribution', description: 'Distributes binary archive.', dependsOn: [mainJar, emfJar, aspectJJar, sourcesJar, javadocJar,
```
Comment (DaGeRe) [inline build.gradle:843]: Why is this commented code left in the source?

File: build.gradle [hunk 6]
```diff
@@ -883,22 +944,22 @@ def sourceContent = project.copySpec {
...
-task distributeSourceTar(type: Tar, group: 'distribution', dependsOn: [mainJar, emfJar, aspectJJar,
-  'kieker-examples:buildStandaloneExamples']) {
-  description = 'Distributes source archive.'
-  archiveClassifier = 'sources'
-  with sourceContent
-
-  compression = Compression.GZIP
-  archiveExtension = 'tar.gz'
-}
+//task distributeSourceZip(type: Zip, group: 'distribution', dependsOn: [mainJar, emfJar, aspectJJar,
```
Comment (DaGeRe) [inline build.gradle:947]: Why is this commented code left in the source?

File: settings.gradle [hunk 7]
```diff
@@ -6,16 +6,31 @@ include 'kieker-monitoring'
...
 include 'kieker-tools:opad'
+include 'kieker-tools:pp-static-log'
+include 'kieker-tools:relabel'
 include 'kieker-tools:resource-monitor'
-//include 'kieker-tools:runtime-analysis'
+include 'kieker-tools:rewrite-log-entries'
+include 'kieker-tools:runtime-analysis'
+include 'kieker-tools:sar'
 include 'kieker-tools:trace-analysis'
-include 'kieker-tools:trace-analysis-new'
```
Comment (DaGeRe) [inline settings.gradle:17]: Why was the order changed here?

**Review by DaGeRe — CHANGES_REQUESTED** [review]
Thanks for the PR. I think one main point is the Java 8/11 thing: With the current deployment, we will also make monitoring unavailable for Java 8. Therefore, we should think about a solution (either adding the missing <T> parts to the methods, or changing the build pipeline). 
Additionally, I would remove the commented out code parts to keep code clean.

File: gradle.properties (hunk 1 above)
Comment (rju) [reply gradle.properties:10]: There where issues with PMD and too low memory settings.

File: gradle.properties (hunk 2 above)
Comment (rju) [reply gradle.properties:17]: 7.0.0-rc4 is the latest version. I had issues with all 6.x pmd versions which appear and disappear at random. However, I can remove this.

File: Jenkinsfile (hunk 3 above)
Comment (rju) [reply Jenkinsfile:34]: We cannot compile most of the analysis when no jdk11 is available. More and more libraries are only compiled for Java11. However, in build.gradle we have set code source and target compatibility for kieker-monitoring to Java8. Thus, it should still be usable with Java8.

File: Jenkinsfile (hunk 4 above)
Comment (rju) [reply Jenkinsfile:101]: Gradle does not know sign as a task and suggest signMavenJavaPublication

File: build.gradle (hunk 5 above)
Comment (rju) [reply build.gradle:843]: As we did not finally discuss to never ever support a zip based binary distribution.

File: build.gradle (hunk 6 above)
Comment (rju) [reply build.gradle:947]: As we did not finally discuss to never ever support a zip based binary distribution.

File: settings.gradle (hunk 7 above)
Comment (rju) [reply settings.gradle:17]: I don't know. Could have been an automatic ordering.

File: Jenkinsfile (hunk 3 above)
Comment (rju) [reply Jenkinsfile:34]: Also there is no API change between Kieker 1.15.x and 2.0.0 in monitoring. Thus, in case of monitoring Java8 is necessary, you can always use older Kieker versions.

**Comment by DaGeRe** [comment]
After the meeting discussion: We upgrade to Java 11 overall, so after this change, I'll approve the PR.
